### PR TITLE
M3-5650 - Remove old DBAAS code

### PR DESCRIPTION
**Pull request by bcmertz**
## Description

Remove old DBAAS code, fix type errors, match API calls

## How to test

Verify no more typescript errors, and that all needed functionality has been preserved

### CODE REVIEW

File: packages/api-v4/src/databases/types.ts [hunk 1]
```diff
@@ -97,16 +100,18 @@ interface ConnectionStrings {
...
-  type: string;
   failover_count: FailoverCount;
+  readonly_count?: ReadonlyCount;
   engine: Engine;
   encrypted: boolean;
   ipv4_public: string;
   ssl_connection: boolean;
   replication_type: ReplicationType;
-  allow_list: string[];
+  allow_list: IPAddress[];
```
Comment (DevDW) [inline packages/api-v4/src/databases/types.ts:114]: The `allow_list` will be an array of strings -- the `IPAddress` type is an object with a number of properties that won't apply here

File: packages/manager/src/hooks/useDatabases.ts [hunk 2]
```diff
@@ -32,10 +33,10 @@ export const useDatabases = () => {
...
   const createDatabase = (payload: CreateDatabasePayload) =>
     dispatch(_create(payload));
-  const deleteDatabase = (databaseID: number) =>
-    dispatch(_delete({ databaseID }));
-  const updateDatabase = (databaseID: number, payload: UpdateDatabasePayload) =>
-    dispatch(_update({ databaseID, ...payload }));
+  const deleteDatabase = ( engine: Engine, databaseID: number) =>
+    dispatch(_delete({ engine, databaseID }));
+  const updateDatabase = (engine: Engine, databaseID: number, payload: UpdateDatabasePayload) =>
+    dispatch(_update({ engine, databaseID, ...payload }));
```
Comment (DevDW) [inline packages/manager/src/hooks/useDatabases.ts:39]: This whole file can be deleted (it's from the previous iteration, when we were using Redux and not RQ)

File: packages/manager/src/store/databases/databases.actions.ts [hunk 3]
```diff
@@ -25,12 +26,15 @@ export const createDatabaseActions = actionCreator.async<
...
-  { databaseID: number },
+  { engine: Engine,
+    databaseID: number
+  },
   {},
   APIError[]
 >(`delete`);
 
 export interface UpdateDatabaseParams extends UpdateDatabasePayload {
+  engine: Engine;
```
Comment (DevDW) [inline packages/manager/src/store/databases/databases.actions.ts:37]: This whole file can be deleted (it's from the previous iteration, when we were using Redux and not RQ)

File: packages/manager/src/store/databases/databases.requests.ts [hunk 4]
```diff
@@ -30,14 +30,15 @@ export const createDatabase = createRequestThunk(
...
 export const deleteDatabase = createRequestThunk(
   deleteDatabaseActions,
-  ({ databaseID }) => _delete(databaseID)
+  ({ engine, databaseID }) => _delete(engine, databaseID)
 );
 
 export const updateDatabase = createRequestThunk(
   updateDatabaseActions,
-  ({ databaseID, ...payload }) => _update(databaseID, payload)
+  ({ engine, databaseID, ...payload }) => _update(engine, databaseID, payload)
```
Comment (DevDW) [inline packages/manager/src/store/databases/databases.requests.ts:41]: This whole file can be deleted (it's from the previous iteration, when we were using Redux and not RQ)

**Review by DevDW — COMMENTED** [review]
Several of these files can still be deleted (noted in the comments), and I believe `DatabaseSettingsLabelPanel.tsx` can be as well

File: packages/manager/src/hooks/useDatabases.ts [hunk 5]
```diff
@@ -32,10 +33,10 @@ export const useDatabases = () => {
   const requestDatabases = () => dispatch(_request({}));
```
Comment (bnussman) [inline packages/manager/src/hooks/useDatabases.ts:33]: this file can be deleted

File: packages/manager/src/store/databases/databases.actions.ts [hunk 6]
```diff
@@ -25,12 +26,15 @@ export const createDatabaseActions = actionCreator.async<
 >(`create`);
 
 export const deleteDatabaseActions = actionCreator.async<
-  { databaseID: number },
+  { engine: Engine,
```
Comment (bnussman) [inline packages/manager/src/store/databases/databases.actions.ts:29]: this file can be deleted

File: packages/manager/src/store/databases/databases.requests.ts [hunk 7]
```diff
@@ -30,14 +30,15 @@ export const createDatabase = createRequestThunk(
   _create
 );
 
+
 export const deleteDatabase = createRequestThunk(
```
Comment (bnussman) [inline packages/manager/src/store/databases/databases.requests.ts:34]: this file can also be deleted

**Comment by bnussman** [comment]
Looking good so far! @bcmertz Can you merge `feature/dbaas` into this PR so we can see if the PR build dev tools are working?

**Review by bnussman — APPROVED** [review]
I didn't look too in depth about how this may conflict with any open PRs, but the clean up looks good to me!

File: packages/api-v4/src/databases/types.ts [hunk 8]
```diff
@@ -94,10 +95,12 @@ interface ConnectionStrings {
 export interface Database {
   id: number;
   label: string;
+  type: string;
+  version: string;
   region: string;
   status: DatabaseStatus;
-  type: string;
   failover_count: FailoverCount;
+  readonly_count?: ReadonlyCount;
```
Comment (DevDW) [inline packages/api-v4/src/databases/types.ts:103]: it's fine to add this now as an optional property, but it is a Phase 2 piece.

**Review by DevDW — APPROVED** [review]
LGTM, no TypeScript or compilation errors 👍

File: packages/manager/src/features/Databases/DatabaseDetail/DatabaseSettings/DatabaseSettings.tsx [hunk 9]
```diff
@@ -1,5 +1,4 @@
 import * as React from 'react';
-// import useDatabases from 'src/hooks/useDatabases';
 // import DatabaseSettingsLabelPanel from './DatabaseSettingsLabelPanel';
 // import DatabaseSettingsMaintenancePanel from './DatabaseSettingsMaintenancePanel';
 // import DatabaseSettingsPasswordPanel from './DatabaseSettingsPasswordPanel';
```
Comment (tiffwong) [inline packages/manager/src/features/Databases/DatabaseDetail/DatabaseSettings/DatabaseSettings.tsx:4]: Is there a reason we're keeping these comments?

**Review by tiffwong — APPROVED** [review]
Left one non-blocking comment but otherwise great job! LGTM 🚀

**Pull request merged**